### ISSUE TITLE
memory leak: dotnet-proc-maps

### ISSUE THREAD

**Issue by mikbras** [deislabs/mystikos#421]
```
2021-06-02T00:42:14.6870406Z ./exec.sh /agent/_work/1/s/build/bin/myst exec-sgx --memory-size=1024m rootfs /app/HelloWorld
2021-06-02T00:42:20.8882551Z === blocks in use: 9360 bytes in 3 blocks
2021-06-02T00:42:20.8882949Z 16 bytes
2021-06-02T00:42:20.8883147Z 0x7fc7043eb89f: myst_debug_malloc()
2021-06-02T00:42:20.8883351Z 0x7fc7043eb9e2: myst_debug_calloc()
2021-06-02T00:42:20.8884616Z 0x7fc704408ca4: calloc()
2021-06-02T00:42:20.8884882Z 0x7fc70440bff9: myst_signal_deliver()
2021-06-02T00:42:20.8885104Z 0x7fc7043f9bca: myst_kill_thread_group()
2021-06-02T00:42:20.8885331Z 0x7fc7043d53b6: _syscall()
2021-06-02T00:42:20.8885519Z 8192 bytes
2021-06-02T00:42:20.8885828Z 0x7fc7043ebbff: myst_debug_posix_memalign()
2021-06-02T00:42:20.8886098Z 0x7fc7043ebce0: myst_debug_memalign()
2021-06-02T00:42:20.8886306Z 0x7fc704408d34: memalign()
2021-06-02T00:42:20.8886557Z 0x7fc7043f8f69: myst_run_thread()
2021-06-02T00:42:20.8886763Z 1152 bytes
2021-06-02T00:42:20.8886946Z 0x7fc7043eb89f: myst_debug_malloc()
2021-06-02T00:42:20.8887157Z 0x7fc7043eb9e2: myst_debug_calloc()
2021-06-02T00:42:20.8887382Z 0x7fc704408ca4: calloc()
2021-06-02T00:42:20.8887576Z 0x7fc7043f9196: _syscall_clone()
2021-06-02T00:42:20.8887779Z 0x7fc7043f9a47: myst_syscall_clone()
2021-06-02T00:42:20.8888002Z 0x7fc7043d2d35: _syscall()
2021-06-02T00:42:20.8888116Z 
2021-06-02T00:42:20.8888271Z *** memory leaks detected
```

**Comment by vtikoo** [comment]
The memory leaks are because the dotnet synchronization thread doesn't respond to SIGKILL, and the main thread exits without being able to successfully terminate it. This seems to be a poll() related issue, and is fixed by the pseudo fork PR #414.

**Comment by vtikoo** [comment]
Closed by #414
